vertx-mysql-client/src/main/java/io/vertx/mysqlclient/impl/MySQLDatabaseMetadata.java

```diff
@@ -0,0 +1,69 @@
+package io.vertx.mysqlclient.impl;
+
+import java.util.Objects;
+
+import io.vertx.sqlclient.spi.DatabaseMetadata;
+
+public class MySQLDatabaseMetadata implements DatabaseMetadata {
+  
+  private final String fullVersion;
+  private final boolean isMariaDB;
+  private final int majorVersion;
+  private final int minorVersion;
+  private final int microVersion;
+  
+  public MySQLDatabaseMetadata(String serverVersion) {
+    fullVersion = serverVersion;
+    isMariaDB = serverVersion.toUpperCase().contains("MARIADB");  
+    String[] versionNumbers;
+    if (!isMariaDB) {
+      // we assume the server version follows ${major}.${minor}.${micro} in https://dev.mysql.com/doc/refman/8.0/en/which-version.html
+      versionNumbers = serverVersion.split("\\.");
+    } else {
+      // server version follows ${junk}-${major}.${minor}.${micro}-MariaDB-${junk}
+      String[] parts = serverVersion.split("-");
+      versionNumbers = parts[1].split("\\.");
+    }
+    majorVersion = Integer.parseInt(versionNumbers[0]);
+    minorVersion = Integer.parseInt(versionNumbers[1]);
+    // we should truncate the possible suffixes here
+    String releaseVersion = versionNumbers[2];
+    int indexOfFirstSeparator = releaseVersion.indexOf("-");
+    if (indexOfFirstSeparator != -1) {
+      // handle unstable release suffixes
+      String releaseNumberString = releaseVersion.substring(0, indexOfFirstSeparator);
+      microVersion = Integer.parseInt(releaseNumberString);
+    } else {
+      microVersion = Integer.parseInt(versionNumbers[2]);
+    }
+  }
+  
+  public boolean isMariaDB() {
+    return isMariaDB;
+  }
+
+  @Override
+  public String productName() {
+    return isMariaDB ? "MariaDB" : "MySQL";
```

Comment:
There're a wide range of database servers supporting MySQL C/S protocol such as Percona, Vitess, TiDB... and their reponse version strings differ much. Don't worry about this, I will fix this in another PR since the version string is also used to control the EOF header flag setting in MySQL.

Reply:
I think we should in the doc clearly tells which database are tested and that other is a best effort support

Reply:
There is something like what you describe in https://vertx.io/docs/vertx-mysql-client/java/#_mysql_and_mariadb_version_support_matrix 

I got a repo for testing basic features with more databases in https://github.com/BillyYccc/vertx-mysql-client-compatibility-testkit which is so-called best effort support